Yank/yank.py

```diff
@@ -672,7 +671,9 @@ def sampler_states(self):
 
     @abc.abstractmethod
     def create(self, thermodynamic_state, sampler_states, storage,
-               unsampled_thermodynamic_states, metadata):
+               unsampled_thermodynamic_states=None,
+               initial_thermodynamic_states=None,
+               metadata=None):
```

Comment:
I wouldn't add default arguments to abstract methods. `AlchemicalPhase` either use them or not so eventual default arguments don't have effect. Also Python doesn't enforce them anyway.

Also, if `AlchemicalPhase` doesn't use `initial_thermodynamic_states` (I haven't found it after a brief search), I would take it away from the contract. This abstract class is only here to document what the user must absolutely implement to have a sampler compatible with `AlchemicalPhase`. Everything else should be up to him.